src/thor/triplegbuilder.cc

```diff
@@ -91,93 +91,190 @@ void AssignAdmins(const AttributesController& controller,
   }
 }
 
+/**
```

Comment:
this function is used to modify the tripleg when we need to add a new incident or update an existing one. let me know if i should move this to the triplegbuilder private header.